src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -92,6 +104,11 @@ public static void RecordException(this Activity activity, Exception ex)
                 tagsCollection.Add(SemanticConventions.AttributeExceptionMessage, ex.Message);
             }
 
+            foreach (var tag in tags)
```

Comment:
Add a `null` check for `activity` in the beginning to `return` if the `activity` is `null` and avoid creating `tagsCollection` and copying from `tagList`.

Reply:
Added null check.

This brought my attention to a somewhat interesting question: what if `ex` is null, but `tags` is not?

One might argue that the only reason someone would call `RecordException` is if they had a non-null exception and that if they just wanted to add an event then they'd call `Activity.AddEvent`.

But what do folks think? Should we add an event anyways even though no [exception related attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md#attributes) may end up on the event?

Reply:
> Should we add an event anyways even though no [exception related attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md#attributes) may end up on the event?

It doesn't sound right to me. I assume most people would use this in the `catch` block or some callback for when an exception occurs. So, it should not be a common scenario. I would say, as you mentioned, they should just use `Activity.AddEvent` if it has nothing to do with an actual exception.